frodo.js

```diff
@@ -3,82 +3,88 @@
  *
```

Comment:
SCRIPT-This function might benefit from using async/await instead of callbacks for better readability.